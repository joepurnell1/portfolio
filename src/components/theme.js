const colours = {
  white: '#FFFFFF',
  accentPink: '#E84855',
  lightPink: '#F7A0A7',
  fairlyDarkBlue: '#022e51',
  darkestBlue: '#012b4c',
  peach: '#FFA987',
  blue: '#067BC2',
  kindaGreen: '#00FFC5',

  // new
  pinkish: '#f77ca0',
  solidPinkish: '#ff5158',
  leblue: '#26ebe3',
};

export default {
  breakpoints: {
    small: '420px',
    medium: '720px',
    large: '1060px',
  },

  typography: {
    headerColour: colours.solidPinkish,
    heavyWeight: 600,
    textColour: colours.darkestBlue,
    smallTextColour: colours.darkestBlue,
    link: {
      standardColour: colours.darkestBlue,
      specialColour: colours.pinkish,
      specialHoverColour: colours.leblue,
      fancyLines: [colours.solidPinkish, '#35FF69', '#440381', '#008DD5', '#F038FF']
    }
  },

  project: {
    accentOneStart: 'rgba(247, 124, 160, 0.4)',
    accentOneEnd: 'rgba(247, 124, 160, 0.7)',
  },

  tag: {
    colours:  [colours.solidPinkish, '#2ee65d', '#440381', '#008DD5', '#F038FF'],
  },

  Global: {
    background: colours.white,
    textColour: colours.darkestBlue,
    strongTextWeight: 600,
  },
  Header: {
    accent: colours.accentPink,
  },
  Link: {
    accent: colours.accentPink,
  },
  Homepage: {
    highlightedBackground: colours.accentPink,
  },
  ColouredInfoBand: {
    backgroundColour: colours.accentPink,
    textColour: colours.white,
  },
  HeadedSection: {
    headerText: colours.accentPink,
  },
};