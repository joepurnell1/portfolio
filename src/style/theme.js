const colours = {
  white: '#FFFFFF',
  accentPink: '#E84855',
  lightPink: '#F7A0A7',
  darkestBlue: '#011627',
  yellow: '#F9DC5C',
};

export default {
  Link: {
    background: colours.white,
    accent: colours.accentPink,
  },
};
