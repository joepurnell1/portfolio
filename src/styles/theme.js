const colours = {
  white: '#FFFFFF',
  accentPink: '#E84855',
  lightPink: '#F7A0A7',
  fairlyDarkBlue: '#022e51',
  darkestBlue: '#011627',
  peach: '#FFA987',
  blue: '#067BC2',
  kindaGreen: '#00FFC5',
};

export default {
  typography: {
    headerColour: colours.accentPink,
    heavyWeight: 600,
    textColour: colours.darkestBlue,
    smallTextColour: colours.accentPink,
  },

  Global: {
    background: colours.white,
    textColour: colours.darkestBlue,
    strongTextWeight: 600,
  },
  Header: {
    accent: colours.accentPink,
  },
  Link: {
    accent: colours.accentPink,
  },
  Homepage: {
    highlightedBackground: colours.accentPink,
  },
  ColouredInfoBand: {
    backgroundColour: colours.accentPink,
    textColour: colours.white,
  },
  HeadedSection: {
    headerText: colours.accentPink,
  },
};
