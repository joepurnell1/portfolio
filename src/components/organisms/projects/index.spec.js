import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Projects from './index';
import theme from '../../../styles/theme';

describe('link component', () => {
  it('should render default as expected', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Projects />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
