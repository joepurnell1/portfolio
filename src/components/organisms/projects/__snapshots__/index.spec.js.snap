// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link component should render default as expected 1`] = `
<div
  className="sc-bxivhb jMTuVJ sc-bwzfXH uTxCW sc-bdVaJa cTPAwo"
>
  <h1
    className="sc-htpNat bxfScw"
  >
    Projects
  </h1>
  <p
    className="sc-ifAKCX dfUAjL"
  >
    Here are some projects I've worked on, both personal and commissioned.
  </p>
  <div
    className="sc-EHOje crurGT"
  >
    <div
      className="sc-bZQynM bNrWxI"
    >
      <h1
        className="sc-htoDjs fPSkLt sc-htpNat jEbsks"
      >
        TraiNap
      </h1>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        An app for train passengers worried about missing their stop.
      </p>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        Sole developer
      </p>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        Built in Java for Android and Swift for iOS
      </p>
      <div
        className="sc-gqjmRU jIroOM"
      >
        <img
          alt="trainap"
          className="sc-jTzLTM eJVCtQ"
          height="190px"
          src="test-file-stub"
          width="auto"
        />
      </div>
    </div>
    <div
      className="sc-bZQynM kiNeCk"
    >
      <img
        alt="doctor"
        className="sc-jTzLTM bLWDQy"
        height="125px"
        src="test-file-stub"
        width="auto"
      />
      <h1
        className="sc-htoDjs bRmxFn sc-htpNat jEbsks"
      >
        DREAMpath
      </h1>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        Commissioned project integrating with Apple's HealthKit
      </p>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        Lead developer
      </p>
      <p
        className="sc-dnqmqq fjHZP sc-ifAKCX dfUAjL"
      >
        Using React Native, Redux, and Styled Components
      </p>
    </div>
    <a
      className="sc-VigVT kpPMkf"
      href="https://github.com/joepurnell1/portfolio"
    >
      <div
        className="sc-gzVnrw fCfsKl sc-bZQynM lepIFy"
      >
        <h1
          className="sc-htoDjs ggMGBI sc-htpNat jEbsks"
        >
          This Website
        </h1>
        <p
          className="sc-iwsKbI iHBDTO sc-ifAKCX dfUAjL"
        >
          A cheeky portfolio, utilising Gatsbyjs and React with a sprinkle of more Styled Components
        </p>
        <img
          alt="website github"
          className="sc-jTzLTM Usefb"
          height="100px"
          src="test-file-stub"
          width="100px"
        />
        <p
          className="sc-gZMcBi iJlxYS"
        >
          See project Repo
        </p>
      </div>
    </a>
  </div>
</div>
`;
